Apps/W1/HybridGP/app/src/Migration/Vendors/GPVendorMigrator.codeunit.al

```diff
@@ -43,6 +43,274 @@ codeunit 4022 "GP Vendor Migrator"
         MigrateVendorAddresses(GPVendor);
     end;
 
+    local procedure MigrateVendorDetails(GPVendor: Record "GP Vendor"; VendorDataMigrationFacade: Codeunit "Vendor Data Migration Facade")
+    var
+        CompanyInformation: Record "Company Information";
+        Vendor: Record Vendor;
+        GenBusinessPostingGroup: Record "Gen. Business Posting Group";
+        VendorPostingGroup: Record "Vendor Posting Group";
+        GPKnownCountries: Record "GP Known Countries";
+        GPPM00200: Record "GP PM00200";
+        GPSY01200: Record "GP SY01200";
+        GPCompanyAdditionalSettings: Record "GP Company Additional Settings";
+        //GPMigrationLog: Record "GP Migration Log";
+        HelperFunctions: Codeunit "Helper Functions";
+        PaymentTermsFormula: DateFormula;
+        VendorNo: Code[20];
+        Country: Code[10];
+        ZipCode: Code[20];
+        ShipMethod: Code[10];
+        PaymentTerms: Code[10];
+        VendorName: Text[50];
+        VendorName2: Text[50];
+        ContactName: Text[50];
+        Address1: Text[50];
+        Address2: Text[50];
+        City: Text[30];
+        State: Text[30];
+        FoundKnownCountry: Boolean;
+        CountryCodeISO2: Code[2];
+        CountryName: Text[50];
+        IsTemporaryVendor: Boolean;
+        HasOpenPurchaseOrders: Boolean;
+        HasOpenTransactions: Boolean;
+        TemporaryVendorMigratedLogTxt: Text[500];
+    begin
+        VendorNo := CopyStr(GPVendor.VENDORID, 1, MaxStrLen(Vendor."No."));
+
+        if not ShouldMigrateVendor(VendorNo, IsTemporaryVendor, HasOpenPurchaseOrders, HasOpenTransactions) then
+            exit;
+
+        if IsTemporaryVendor then begin
+            TemporaryVendorMigratedLogTxt := 'Temporary vendor was migrated because it ';
+
+            if HasOpenPurchaseOrders and HasOpenTransactions then
+                TemporaryVendorMigratedLogTxt += 'has open POs and open AP transactions.'
+            else begin
+                if HasOpenPurchaseOrders then
+                    TemporaryVendorMigratedLogTxt += 'has open POs.';
+
+                if HasOpenTransactions then
+                    TemporaryVendorMigratedLogTxt += 'has open AP transactions.';
+            end;
+
+            //Uncomment when Migration Log feature is finally merged by MSFT
+            //GPMigrationLog.InsertLog(MigrationLogAreaTxt, VendorNo, TemporaryVendorMigratedLogTxt);
```

Comment:
This section will be uncommented when the latest MSFT PR gets merged, which includes the new Migration Log table.